test/extended/router/gatewayapicontroller.go

```diff
@@ -324,12 +315,12 @@ func createAndCheckGateway(oc *exutil.CLI, gwname, gwclassname, domain string) (
 }
 
 func checkGatewayStatus(oc *exutil.CLI, gwname, ingressNameSpace string) (*gatewayapiv1.Gateway, error) {
-	gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+	var err error
 	gateway := &gatewayapiv1.Gateway{}
 
 	waitErr := wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 10*time.Minute, false, func(context context.Context) (bool, error) {
-		gateway, errGwStatus := gwapiClient.GatewayV1().Gateways(ingressNameSpace).Get(context, gwname, metav1.GetOptions{})
-		if errGwStatus != nil {
+		gateway, err = oc.AdminGatewayApiClient().GatewayV1().Gateways(ingressNameSpace).Get(context, gwname, metav1.GetOptions{})
```

Comment:
Should not use `:=` here, otherwise the `gateway.Name` is empty in logs as below
```
I0422 15:27:59.859132 34018 gatewayapicontroller.go:352] Gateway "" successfully programmed!
```

Reply:
@melvinjoseph86 please check your enhancement PR, not sure why you revert the fix.